lib/extensions.js

```diff
@@ -0,0 +1,75 @@
+// Imports
+const fs = require('fs')
+const path = require('path')
+const foundationExtensions = require('../app/config').foundationExtensions || ['govuk-frontend']
+
+// Utils
+const pathFromRoot = (...all) => path.join.apply(null, [__dirname, '..'].concat(all))
+const pathToHookFile = packageName => pathFromRoot('node_modules', packageName, 'govuk-prototype-kit.config.json')
+const objectMap = (object, mapFn) => Object.keys(object).reduce((result, key) => {
+  result[key] = mapFn(object[key], key)
+  return result
+}, {})
+const removeDuplicates = arr => arr.filter((i, pos, self) => self.indexOf(i) === pos)
+const filterOutTreversesAndEmpty = part => part && part !== '..'
+const cloneArray = arr => arr.slice(0)
+
+// Reused Functions
+const lookupConfigForPackage = packageName => {
+  if (fs.existsSync(pathToHookFile(packageName))) {
+    return require(pathToHookFile(packageName))
+  } else {
+    return {}
+  }
+}
+
+const getPackageNamesInOrder = () => {
+  const dependencies = require(pathFromRoot('package.json')).dependencies || {}
+  const allNpmDependenciesInAlphabeticalOrder = Object.keys(dependencies).sort()
+  const installedTopPriorityDependencies = foundationExtensions
+    .filter(packageName => allNpmDependenciesInAlphabeticalOrder.indexOf(packageName) > -1)
+
+  return removeDuplicates(installedTopPriorityDependencies.concat(allNpmDependenciesInAlphabeticalOrder))
+}
+
+const getSourceFromItem = item => typeof item === 'string' ? item : item.src
+
+const extensionItemsByType = getPackageNamesInOrder()
```

Comment:
I find it really hard to follow this (down to line 41) can it be broken up?

Reply:
These are reasonably low-level functions compared with the rest of the file, should we move them into a different space or to the bottom of the file to give the focus to the actual extension workings?